components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -428,4 +447,34 @@ private void setResponseHeaders(OAuthTokenReqMessageContext tokReqMsgCtx,
             tokenRespDTO.setResponseHeaders((ResponseHeader[]) tokReqMsgCtx.getProperty(OAuthConstants.RESPONSE_HEADERS_PROPERTY));
         }
     }
+
+    /**
+     * Method to check whether id token contains the required claims(iss,sub,aud,exp,iat) defined by the oidc spec
+     * @param jwt id token
+     * @return true or false(whether id token contains the required claims)
+     * @throws ParseException
+     */
+    private boolean isValidIdToken(JWT jwt) throws ParseException {
```

Comment:
Since this method is responsible for actually validating the token, we should not have a ParseException in the method signature. The return boolean status can be used instead.